### PR TITLE
Remove redundant hint when already used

### LINKED ISSUE #3365: Remove redundant hint if user is already using the hint
## Description

If users use `kedro new --addons all --name all --example y` then the message:

```
To skip the interactive flow you can run `kedro new` with
kedro new --name=<your-project-name> --addons=<your-addons> --example=<yes/no>
```

Should not appear at the end. See the red highlight below:

image

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Description

Resolves #3365 
Found that "You have selected no project tools" wasn't always shown when appropriate, included that fix in this PR

## Development notes
Instead of testing the negative case, the tests test that the hint is shown where relevant. Manual testing reveals the hint is no longer shown where already used. Unsure on best practice here - happy to include a unit test for the negative case as well.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by merelcht** [comment]
> Instead of testing the negative case, the tests test that the hint is shown where relevant. Manual testing reveals the hint is no longer shown where already used. Unsure on best practice here - happy to include a unit test for the negative case as well.

IMO it would be good to also add the negative case, see e.g. https://github.com/kedro-org/kedro/pull/3424/files

File: kedro/framework/cli/starters.py [hunk 1]
```diff
@@ -299,6 +299,13 @@ def new(  # noqa: PLR0913
 
     _create_project(project_template, cookiecutter_args)
 
+    if prompts_required and not config_path:
```
Comment (DimedS) [inline kedro/framework/cli/starters.py:302]: Is it true that it will be shown when using with starter? Where no tools allowed?

Comment (ankatiyar) [inline kedro/framework/cli/starters.py:302]: Should also not show the hint if `--starter` is used

Comment (AhdraMeraliQB) [reply kedro/framework/cli/starters.py]: Addressed in https://github.com/kedro-org/kedro/pull/3439/commits/8f86cf86ee49948e79a9bf2afe8abd2d58543a5f

Comment (AhdraMeraliQB) [reply kedro/framework/cli/starters.py]: Addressed in https://github.com/kedro-org/kedro/pull/3439/commits/8f86cf86ee49948e79a9bf2afe8abd2d58543a5f

File: tests/framework/cli/test_starters.py [hunk 2]
```diff
@@ -1293,6 +1326,10 @@ def test_valid_tools_flag(self, fake_kedro_cli, tools, example_pipeline):
         tools = _convert_tool_names_to_numbers(selected_tools=tools)
         _assert_template_ok(result, tools=tools, example_pipeline=example_pipeline)
         _assert_requirements_ok(result, tools=tools, repo_name="new-kedro-project")
+        assert (
+            "To skip the interactive flow you can run `kedro new` with\nkedro new --name=<your-project-name> --tools=<your-project-tools> --example=<yes/no>"
```
Comment (stichbury) [inline tests/framework/cli/test_starters.py:1344]: It would make for more text to read but I think it may help to give an example that the user can copy e.g. `kedro new --name=example-project --tools=none -example=no`

File: tests/framework/cli/test_starters.py [hunk 3]
```diff
@@ -854,6 +854,20 @@ def test_git_repo_custom_directory(
...
 
+    def test_no_hint(self, fake_kedro_cli):
+        shutil.copytree(TEMPLATE_PATH, "template")
+        result = CliRunner().invoke(
+            fake_kedro_cli,
+            ["new", "-v", "--starter", str(Path("./template").resolve())],
+            input=_make_cli_prompt_input(),
+        )
+        assert (
+            "To skip the interactive flow you can run `kedro new` with\nkedro new --name=<your-project-name> --tools=<your-project-tools> --example=<yes/no>"
```
Comment (AhdraMeraliQB) [inline tests/framework/cli/test_starters.py:865]:
```suggestion
            "To skip the interactive flow you can run `kedro new` with\nkedro new --name=example-project --tools=none --example=no"
```

**Review by DimedS — APPROVED** [review]
Thank you for the PR @AhdraMeraliQB , great job!

File: tests/framework/cli/test_starters.py (hunk 2 above)
Comment (AhdraMeraliQB) [reply tests/framework/cli/test_starters.py:1344]: Resolved offline

**Review by merelcht — APPROVED** [review]
LGTM! Thanks for adding the extra tests, this makes it clear when the message is expected and when it isn't 👍
